realm/src/main/java/io/realm/Realm.java

```diff
@@ -123,6 +126,12 @@
 public final class Realm implements Closeable {
     public static final String DEFAULT_REALM_NAME = "default.realm";
 
+    // This single thread executor ensures that only one finalizer thread ever exists
+    private static final ExecutorService executorService = Executors.newSingleThreadExecutor();
+
+    // This does not need to be thread safe since it's only used in a synchronized method
+    private static boolean isFinalizerStarted = false;
```

Comment:
I think this should be `static volatile` to avoid other Threads caching the value